src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -896,63 +896,67 @@ Namespace Microsoft.CodeAnalysis.Semantics
             Return New LazyRelationalCaseClause(value, relation, CaseKind, _semanticModel, syntax, type, constantValue, isImplicit)
         End Function
 
-        Private Function CreateBoundDoLoopStatementOperation(boundDoLoopStatement As BoundDoLoopStatement) As IWhileUntilLoopStatement
+        Private Function CreateBoundDoLoopStatementOperation(boundDoLoopStatement As BoundDoLoopStatement) As IDoLoopStatement
+            Dim doLoopKind As DoLoopKind = GetDoLoopKind(boundDoLoopStatement)
             Dim isTopTest As Boolean = boundDoLoopStatement.ConditionIsTop
             Dim isWhile As Boolean = Not boundDoLoopStatement.ConditionIsUntil
             Dim condition As Lazy(Of IOperation) = New Lazy(Of IOperation)(Function() Create(boundDoLoopStatement.ConditionOpt))
```

Comment:
>boundDoLoopStatement.ConditionOpt [](start = 93, length = 33)

It looks like we are going to drop the other condition in error cases, both top and bottom conditions could be present.

Reply:
Would it be better to expose both conditions?

---
In reply to: [135968820](https://github.com/dotnet/roslyn/pull/21791#discussion_r135968820) [](ancestors = 135968820)

Reply:
Done

---
In reply to: [135968884](https://github.com/dotnet/roslyn/pull/21791#discussion_r135968884) [](ancestors = 135968884,135968820)